Technical-Steering-Committee/guidelines/breaking-changes.md

```diff
@@ -0,0 +1,136 @@
+# Breaking Changes Policy
+
+Having a reliable platform that can evolve is vital for the success of Zowe. We need to make the adoption of new features and change easy and smooth for the users and extenders.
+
+Breaking changes are sometimes necessary to keep the Zowe project in a good shape. This document provides guidelines and policy about what is breaking change, how it can be introduced so the change will have minimal disruptive effects on the Zowe users and extenders.
+
+This policy will be affective with the Zowe V2 release. It will affect how changes to V2 are introduced. Breaking changes to V2 APIs and behavior can be made in V3 at earliest.
+
+## Public and internal API
+
+The term API in this document means any type of programmatic API including REST APIs, Java APIs, TypeScript APIs, and CLIs.
+
+All functionality in the official Zowe documentation is part of the public API.
+
+Any undocumented command, endpoint, object, property, method, argument, behavior, or event is internal API.
+
+Exceptions are undocumented behaviors that are relied on by the community (e.g. `zowe jobs submit` command returns the job ID without other data or messages). Such occurrences should be reported via issues and documented over time.
+
+Sometimes, there is disagreement about whether functionality is internal or public. In those cases, the TSC makes a determination.
+
+## Breaking changes
+
+See [Zowe Version Release Schedule](/Technical-Steering-Committee/release.md) for Zowe terminology about releases.
+
+![Schedule for the nearest releases](/Technical-Steering-Committee/major_releases.png)
+
+At least two TSC members must approve backward-incompatible changes to the under development branch or current active or maintenance release. Changes to the development branch will be breaking changes in the upcoming active release. Breaking changes to active or maintenance release should be avoided unless necessary (for example critical fixes or security vulnerabilities).
```

Comment:
I would be careful about the under development branch. It's possible that some squads develop in under development branch including breaking changes and cherry-pick to the active and major branches.